sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/AsyncProgressTrackingMicroBatchExecutionSuite.scala

```diff
@@ -750,6 +750,12 @@ class AsyncProgressTrackingMicroBatchExecutionSuite
             }
             e.getCause.getCause.getMessage should include("Permission denied")
           }
+      },
+      Execute { _ =>
+        // SPARK-41894: Restore the write permission of `commitDir`
+        // so that `mvn clean` can run successfully.
+        val commitDir = new File(checkpointLocation + path)
+        commitDir.setWritable(true)
```

Comment:
https://github.com/apache/spark/blob/58d5dc3d573dfbb6d21ea41d101550146756f45b/sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/AsyncProgressTrackingMicroBatchExecutionSuite.scala#L736-L743

Line 741 `commitDir.setReadOnly()`

Reply:
Is this ok? Or put the new action to finally block?

Reply:
I'd prefer restoring the permission in finally block.

Reply:
[7716abc](https://github.com/apache/spark/pull/39406/commits/7716abc32aa3b070e52b1e0f087a5fcc089c0ac3) change to use `finally block`